### PR TITLE
Feature/td ticket20 fuel backend

### LINKED ISSUE #20: As a user, I want to be able to estimate the carbon costs of my combustion fuel intake
**__User Story Description__**
As a user
I want to be able to estimate the costs of my combustion fuel intake
So I can inform the cost of my use?

**__Steps to Follow (optional)__**
- [ ] Research calculations needed for Carbon API
- [ ] Communicate to frontend what questions need to be added to UI
- [ ] Fuel Model
- [ ] Fuel Schema
- [ ] Fuel resolver
- [ ] Additional steps as necessary

**__Additional Considerations__**
After determining what data is need for Carbon API request, communicate this with Issue #42

### PR DESCRIPTION
**Pull request by TiciaD**
- setup schema for fuel response/request
- setup router for fuel endpoint
- added fuel endpoint to tRPC router

Closes #20

### CODE REVIEW

File: src/schema/fuel.schema.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const fuelSourceUnitSchema = z.enum(['gallon', 'btu', 'thousand_cubic_feet']);
+const fuelSourceTypeSchema = z.enum(['dfo', 'ng', 'pg', 'ker']);
+
+export const unregisteredFuelRequestSchema = z.object({
+  fuel_source_type: fuelSourceTypeSchema,
+  fuel_source_unit: fuelSourceUnitSchema,
+  fuel_source_value: z.number(),
+});
+
+export type fuelSourceUnit = z.TypeOf<typeof fuelSourceUnitSchema>;
```
Comment (ialej001) [inline src/schema/fuel.schema.ts]: Minor nitpick but exported types should start with a capital letter as well. Ex: fuelSourceUnit -> FuelSourceUnit

File: src/server/router/index.ts [hunk 2]
```diff
@@ -5,12 +5,14 @@ import superjson from 'superjson';
...
 import { electricityRouter } from './electricity.router';
+import { fuelRouter } from './fuel.router';
 
 export const appRouter = createRouter()
   .transformer(superjson)
   .merge('example.', exampleRouter)
   .merge('question.', protectedExampleRouter)
-  .merge('electricity.', electricityRouter);
+  .merge('electricity.', electricityRouter)
+  .merge('fuel', fuelRouter);
```
Comment (ialej001) [inline src/server/router/index.ts]: Minor nitpick, you missed the '.' after fuel.

**Review by ialej001 — CHANGES_REQUESTED** [review]
Literally minor nitpicks on some conventions that should be observed for consistency :)

File: src/schema/fuel.schema.ts (hunk 1 above)
Comment (TiciaD) [reply src/schema/fuel.schema.ts]: Added capitalization to this and forgot to do this in the electricity schema so fixed that as well!

File: src/server/router/index.ts (hunk 2 above)
Comment (TiciaD) [reply src/server/router/index.ts]: Added period, thanks for catching that!

**Review by ialej001 — APPROVED** [review]

**Review by victoriacheng15 — APPROVED** [review]
I saw defined types for fuel inputs and names are consistent.

**Pull request merged**